static/app/views/issueList/groupSearchViewTabs/draggableTabBar.tsx

```diff
@@ -85,106 +97,230 @@ export function DraggableTabBar({
+      ? {
+          id: 'temporary-tab',
+          key: 'temporary-tab',
+          label: 'Unsaved',
+          query: queryParams.query,
+          querySort: queryParams.sort,
+          queryCount: undefined,
+        }
+      : undefined
+  );
+
+  const handleOnReorder = (newOrder: Node<DraggableTabListItemProps>[]) => {
+    const newTabs = newOrder
+      .map(node => {
+        const foundTab = tabs.find(tab => tab.key === node.key);
+        return foundTab?.key === node.key ? foundTab : null;
+      })
+      .filter(defined);
+    setTabs(newTabs);
+    onReorder?.(newTabs);
+  };
+
+  const handleOnSaveChanges = () => {
+    const originalTab = tabs.find(tab => tab.key === selectedTabKey);
+    if (originalTab) {
+      const newTabs = tabs.map(tab => {
+        if (tab.key === selectedTabKey && tab.unsavedChanges) {
+          tab.query = tab.unsavedChanges[0];
+          tab.querySort = tab.unsavedChanges[1];
+          tab.unsavedChanges = undefined;
+        }
+        return tab;
+      });
+      setTabs(newTabs);
+      onSave?.(newTabs);
     }
-  }, [showTempTab, selectedTabKey, tabs]);
+  };
 
-  const onReorder: (newOrder: Node<DraggableTabListItemProps>[]) => void = newOrder => {
-    setTabs(
-      newOrder
-        .map(node => {
-          const foundTab = tabs.find(tab => tab.key === node.key);
-          return foundTab?.key === node.key ? foundTab : null;
+  const handleOnDiscardChanges = () => {
+    const originalTab = tabs.find(tab => tab.key === selectedTabKey);
+    if (originalTab) {
+      setTabs(
+        tabs.map(tab => {
+          if (tab.key === selectedTabKey) {
+            tab.unsavedChanges = undefined;
+          }
+          return tab;
         })
-        .filter(defined)
-    );
+      );
+      router.push(
+        normalizeUrl({
+          query: {
+            ...queryParams,
+            query: originalTab.query,
+            sort: originalTab.querySort,
+            ...(originalTab.id ? {viewId: originalTab.id} : {}),
+          },
+          pathname: `/organizations/${orgSlug}/issues/`,
+        })
+      );
+      onDiscard?.();
+    }
   };
 
-  const handleOnDelete = (tab: Tab) => {
-    if (tabs.length > 1) {
-      setTabs(tabs.filter(tb => tb.key !== tab.key));
-      onDelete?.(tab.key);
+  const handleOnTabRenamed = (newLabel: string, tabKey: string) => {
+    const renamedTab = tabs.find(tb => tb.key === tabKey);
+    if (renamedTab && newLabel !== renamedTab.label) {
+      const newTabs = tabs.map(tab =>
+        tab.key === renamedTab.key ? {...tab, label: newLabel} : tab
+      );
+      setTabs(newTabs);
+      onTabRenamed?.(newTabs, newLabel);
     }
   };
 
-  const handleOnDuplicate = (tab: Tab) => {
-    const idx = tabs.findIndex(tb => tb.key === tab.key);
+  const handleOnDuplicate = () => {
+    const idx = tabs.findIndex(tb => tb.key === selectedTabKey);
     if (idx !== -1) {
-      setTabs([
+      const tempId = generateTempViewId();
```

Comment:
So I made it this the "_xxxxx" format to avoid any sudden, noticeable changes when the viewId is eventually populated in the URL. I figured it might be strange to see the query parameter jump from 12-3-4-5-678 to something like 50001, but I also could be overthinking this lol

Reply:
Ah yeah that's true, the full UUID might be a bit jarring. This does have a small chance of creating a conflicting entry, but it's probably so unlikely that it's not worth thinking about. Though you could just have an incrementing index like `_1`, `_2` which wouldn't have the same problem. Just food for thought

Reply:
haha yeah i thought about that too - i also think this should be fine, especially since the id is temporary and is likely to be replaced in <5 seconds, I think we should be in the clear.